llvm/lib/Target/DirectX/DXILFlattenArrays.cpp

```diff
@@ -321,6 +321,18 @@ static void collectElements(Constant *Init,
     Elements.push_back(Init);
     return;
   }
+  auto *LLVMArrayType = dyn_cast<ArrayType>(Init->getType());
```

Comment:
This should either be `cast<ArrayType>` (since the cast can't fail) or we should do the `dyn_cast` and check for null above instead of `isa`.

This is also kind of a weird name - I think `ArrayTy` is the usual abbreviation.